LocalDriver (Chrome, Firefox) required on local machine but RemoteDriver is used

For project which is running tests remotely using SeleniumGrid, I have configured ChromeDriver as an optional server and even I'm not using it FluentLenium is trying to create an instance of the driver and throwing errors (not running tests on Selenium Grid)

```
10:37:55.269 Caused by: java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://github.com/SeleniumHQ/selenium/wiki/ChromeDriver. The latest version can be downloaded from http://chromedriver.storage.googleapis.com/index.html
10:37:55.269 	at com.google.common.base.Preconditions.checkState(Preconditions.java:199)
10:37:55.269 	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:109)
``` 

So it requires to have ChromeDriver or Firefox installed but doesn't really need it on the machine where the Selenium tests are running because these tests would be run remotely.

--- a/README.md
+++ b/README.md
@@ -897,8 +897,8 @@ constructor matching parent class, and generic type should be a hook option clas
 
 ```
 public class ExampleHook extends BaseHook<ExampleHookOptions> {
-    public ExampleHook(FluentControl fluentControl, ComponentInstantiator instantiator, Supplier<WebElement> elementSupplier, Supplier<ElementLocator> locatorSupplier, ExampleHookOptions options) {
-        super(fluentControl, instantiator, elementSupplier, locatorSupplier, options);
+    public ExampleHook(FluentControl control, ComponentInstantiator instantiator, Supplier<WebElement> elementSupplier, Supplier<ElementLocator> locatorSupplier, ExampleHookOptions options) {
+        super(control, instantiator, elementSupplier, locatorSupplier, options);
     }
 }
 

--- a/fluentlenium-core/src/main/java/org/fluentlenium/adapter/FluentAdapter.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/adapter/FluentAdapter.java
@@ -7,6 +7,7 @@ import org.fluentlenium.configuration.ConfigurationProperties;
 import org.fluentlenium.configuration.WebDrivers;
 import org.fluentlenium.core.FluentControl;
 import org.fluentlenium.core.FluentDriver;
+import org.fluentlenium.core.SeleniumDriverControl;
 import org.fluentlenium.core.inject.ContainerContext;
 import org.fluentlenium.core.inject.ContainerFluentControl;
 import org.openqa.selenium.WebDriver;
@@ -43,7 +44,7 @@ public class FluentAdapter implements FluentControl, ConfigurationProperties {
         return configuration;
     }
 
-    @Delegate(types = FluentControl.class)
+    @Delegate(types = FluentControl.class, excludes = SeleniumDriverControl.class) // We want getDriver to be final.
     private ContainerFluentControl getFluentControl() {
         return (ContainerFluentControl) getDriverContainer().getFluentControl();
     }
@@ -54,6 +55,11 @@ public class FluentAdapter implements FluentControl, ConfigurationProperties {
 
     private void setFluentControl(ContainerFluentControl fluentControl) {
         getDriverContainer().setFluentControl(fluentControl);
+    }
+
+    @Override
+    public final WebDriver getDriver() {
+        return getFluentControl().getDriver();
     }
 
     protected FluentControlContainer getDriverContainer() {

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/DefaultFluentContainer.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/DefaultFluentContainer.java
@@ -7,7 +7,7 @@ import lombok.experimental.Delegate;
  */
 public class DefaultFluentContainer implements FluentControl, FluentContainer {
 
-    private FluentControl control;
+    protected FluentControl control;
 
     public DefaultFluentContainer() {
     }

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/FluentControl.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/FluentControl.java
@@ -13,6 +13,6 @@ import org.fluentlenium.core.search.SearchControl;
 import org.fluentlenium.core.snapshot.SnapshotControl;
 import org.fluentlenium.core.wait.AwaitControl;
 
-public interface FluentControl extends SearchControl<FluentWebElement>, AwaitControl, InputControl, JavascriptControl, AlertControl, SnapshotControl, EventsControl, NavigationControl, CssControl, FluentInjectControl, ComponentInstantiator {
+public interface FluentControl extends SearchControl<FluentWebElement>, AwaitControl, InputControl, JavascriptControl, AlertControl, SnapshotControl, EventsControl, NavigationControl, SeleniumDriverControl, CssControl, FluentInjectControl, ComponentInstantiator {
 
 }

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/SeleniumDriverControl.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/SeleniumDriverControl.java
@@ -1,0 +1,18 @@
+package org.fluentlenium.core;
+
+import org.fluentlenium.adapter.FluentAdapter;
+import org.openqa.selenium.WebDriver;
+
+public interface SeleniumDriverControl {
+    /**
+     * Get the actual underlying Selenium WebDriver.
+     * <p>
+     * To customize the WebDriver, you should configure {@link org.fluentlenium.adapter.FluentAdapter#getWebDriver()}
+     * or override {@link FluentAdapter#newWebDriver()}.
+     * <p>
+     * This method can't be override to customize the WebDriver.
+     *
+     * @return The actual underlying Selenium WebDriver
+     */
+    WebDriver getDriver();
+}

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/domain/Component.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/domain/Component.java
@@ -8,12 +8,12 @@ import org.openqa.selenium.WebElement;
 public class Component {
     @NoInject
     protected WebElement webElement;
-    protected FluentControl fluentControl;
+    protected FluentControl control;
     protected ComponentInstantiator instantiator;
 
-    public Component(WebElement webElement, FluentControl fluentControl, ComponentInstantiator instantiator) {
+    public Component(WebElement webElement, FluentControl control, ComponentInstantiator instantiator) {
         this.webElement = webElement;
-        this.fluentControl = fluentControl;
+        this.control = control;
         this.instantiator = instantiator;
     }
 }

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/domain/FluentWebElement.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/domain/FluentWebElement.java
@@ -58,12 +58,12 @@ public class FluentWebElement extends Component implements WrapsElement, FluentA
     public FluentWebElement(WebElement webElement, FluentControl fluentControl, ComponentInstantiator instantiator) {
         super(webElement, fluentControl, instantiator);
 
-        this.hookChainBuilder = new DefaultHookChainBuilder(this.fluentControl, this.instantiator);
+        this.hookChainBuilder = new DefaultHookChainBuilder(this.control, this.instantiator);
 
         this.search = new Search(webElement, this.instantiator);
         this.axes = new Axes(webElement, this.instantiator, this.hookChainBuilder);
-        this.mouseActions = new MouseElementActions(this.fluentControl.getDriver(), webElement);
-        this.keyboardActions = new KeyboardElementActions(this.fluentControl.getDriver(), webElement);
+        this.mouseActions = new MouseElementActions(this.control.getDriver(), webElement);
+        this.keyboardActions = new KeyboardElementActions(this.control.getDriver(), webElement);
         this.conditions = new WebElementConditions(this);
         this.label = new FluentLabelImpl<>(this, new Supplier<String>() {
             @Override
@@ -75,7 +75,7 @@ public class FluentWebElement extends Component implements WrapsElement, FluentA
 
     @Delegate(excludes = {InputControl.class, AwaitControl.class, SearchControl.class})
     private FluentControl getFluentControl() {
-        return fluentControl;
+        return control;
     }
 
     /**
@@ -124,7 +124,7 @@ public class FluentWebElement extends Component implements WrapsElement, FluentA
     }
 
     public FluentWaitElement await() {
-        return new FluentWaitElement(fluentControl.await(), this);
+        return new FluentWaitElement(control.await(), this);
     }
 
     /**
@@ -281,7 +281,7 @@ public class FluentWebElement extends Component implements WrapsElement, FluentA
      */
 
     public boolean clickable() {
-        return ExpectedConditions.elementToBeClickable(getElement()).apply(fluentControl.getDriver()) != null;
+        return ExpectedConditions.elementToBeClickable(getElement()).apply(control.getDriver()) != null;
     }
 
     /**
@@ -290,7 +290,7 @@ public class FluentWebElement extends Component implements WrapsElement, FluentA
      * @return false is the element is still attached to the DOM, true otherwise.
      */
     public boolean stale() {
-        return ExpectedConditions.stalenessOf(getElement()).apply(fluentControl.getDriver());
+        return ExpectedConditions.stalenessOf(getElement()).apply(control.getDriver());
     }
 
     /**

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/hook/BaseHook.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/hook/BaseHook.java
@@ -34,8 +34,8 @@ public class BaseHook<T> extends DefaultFluentContainer implements FluentHook<T>
         return locatorSupplier.get();
     }
 
-    public BaseHook(FluentControl fluentControl, ComponentInstantiator instantiator, Supplier<WebElement> elementSupplier, Supplier<ElementLocator> locatorSupplier, Supplier<String> toStringSupplier, T options) {
-        super(fluentControl);
+    public BaseHook(FluentControl control, ComponentInstantiator instantiator, Supplier<WebElement> elementSupplier, Supplier<ElementLocator> locatorSupplier, Supplier<String> toStringSupplier, T options) {
+        super(control);
         this.instantiator = instantiator;
         this.elementSupplier = elementSupplier;
         this.locatorSupplier = locatorSupplier;

--- a/fluentlenium-core/src/main/java/org/fluentlenium/core/navigation/NavigationControl.java
+++ b/fluentlenium-core/src/main/java/org/fluentlenium/core/navigation/NavigationControl.java
@@ -5,7 +5,6 @@ import org.fluentlenium.core.action.WindowAction;
 import org.fluentlenium.core.domain.FluentList;
 import org.fluentlenium.core.domain.FluentWebElement;
 import org.openqa.selenium.Cookie;
-import org.openqa.selenium.WebDriver;
 
 import java.util.Set;
 
@@ -96,11 +95,4 @@ public interface NavigationControl {
      * @return current URL
      */
     String url();
-
-    /**
-     * Return the raw Selenium WebDriver
-     *
-     * @return selenium webdriver
-     */
-    WebDriver getDriver();
 }

